### DISCUSSION TITLE
airflow db init -> illegal instruction

### DISCUSSION THREAD

**Discussion by dandumit** [apache/airflow#21355] [General]
Hello All,
I am struggling to install and test Airflow. 
When I am reaching point : airflow db init : 

[2022-02-06 12:37:49,782] {db.py:921} INFO - Creating tables
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
Illegal instruction

I cannot find error on logs. what should I do ? 
Thank you,
Daniel

**Comment by potiuk** [comment]
`Illegal instruction` -> check if you are fulfilling all requirements. Illegal instruction indicate that you might have a problem with your platform. I am guessing - because you forgot to provide any details of your installation -  If you are running on MacOS M1 - this is likely you have problem with some dependencies that are not ARM compatible. Airflow does not yet support M1 MacOS (not all dependencies are supported). You might follow #15635 to be notified when it happens.

**Comment by dandumit** [comment]
Thank you @potiuk  - you are all around.
It's on a synology DSM , I am really struggling to get it started.  I have run for prerequisites but still this error. 
I have tried to configure it to use MySQL. 
WHat else should I try  ?
Thank you,
Daniel

**Comment by potiuk** [comment]
Actually the problem might be the processor you have. Look here:

https://tech.amikelive.com/node-887/how-to-resolve-error-illegal-instruction-core-dumped-when-running-import-tensorflow-in-a-python-program/

Seems that this error will happen if you have an OLD machine that does not have the right instruction set in its processor. And I think I do recall that some of the Mac-Minis even those rather recently released (like 6 or 7 years ago) actually had a very old processor that might have caused the issue. 

So if you happen to run it on a Mac Mini or a really old processor, the solution is easy. Change your hardware.

**Reply by potiuk** [reply]
Ahhhh So you run on Synology - then that's it . Probably they DO use an old processor.

**Reply by potiuk** [reply]
If so then the only way I could recommend you is you try to build the image on this processor's docker and possibly during the build the libraries will detect the lack of support of those instructions and will build a version that is not as optimized but without using the "newish" instructions. But most likely than not the build will fail and you will have to tweak it to get it build. 

Good luck :).
